HomeWork1/Novikov_Homework_2.cpp

```diff
@@ -0,0 +1,283 @@
+  std::cin >> number;
+
+  while (number) {
+    reverse_number = kTen * reverse_number + number % 10;
+    number /= 10;
+  }
+
+  std::cout << "Result is: " << reverse_number << std::endl;
+}
+
+void Task_4_odd_numbers() {
+  constexpr int kRangeArraySizeLow{1};
+  constexpr int kRangeArraySizeHigh{50};
+  constexpr int kRangeArrayElementLow{-60};
+  constexpr int kRangeArrayElementHigh{90};
+
+  int elementsNumber{0};
+  int elements[50]{0};
+  bool check{true};
+
+  while (check) {
+    std::cout << "Enter number of elements [1..50]: ";
+    std::cin >> elementsNumber;
+    if (elementsNumber < kRangeArraySizeLow ||
+        elementsNumber > kRangeArraySizeHigh) {
+      std::cout << "Number is not in range! Please try again\n";
+      continue;
+    }
+    check = false;
+  }
+
+  std::cout << "Enter elements in range [-60..90]\n";
+  for (int i = 0; i < elementsNumber;) {
+    std::cin >> elements[i];
+    if (elements[i] <= kRangeArrayElementLow ||
+        elements[i] >= kRangeArraySizeHigh) {
+      std::cout << "Element is not in range! Please try again\n";
+      continue;
+    }
+    i++;
+  }
+
+  int oddSum{0};
+
+  for (int i = 0; i < elementsNumber; i++)
+    if (elements[i] & 1)
+      oddSum += elements[i];
+
+  std::cout << "Sum of odd elements: " << oddSum << std::endl;
+}
+
+int sum_i(int number) {
+  if (number < 10)
+    return number;
+  else {
+    int sum{0};
+    sum = sum_counter(number);
+    return sum;
+  }
+}
+
+void Task_5_best_divider() {
+
+  int number{0};
+
+  std::cout << "Enter a number ";
+  std::cin >> number;
+
+  if (number < 0)
+    number *= -1;
+
+  int best = 0;
+  std::cout << "Dividers = { ";
+  for (int i = 1; i <= number; i++)
+    if (number % i == 0) {
+      if (sum_i(best) <= sum_i(i))
+        best = i;
+      std::cout << i << " ";
+    }
+  std::cout << "}\n";
+
+  std::cout << "The best divider is: " << best << std::endl;
+}
+
+void Task_6_christmas_tree() {
+  int number{0};
+  int star{1};
+
+  std::cout << "Enter number: ";
+  std::cin >> number;
+
+  int spaces = number / 2;
+  while (star <= number + 1) {
+    for (int i = 0; i < spaces; i++)
+      std::cout << " ";
+    for (int i = 0; i < star; i++)
+      std::cout << "*";
+    std::cout << std::endl;
+    star += 2;
+    spaces--;
+  }
+}
+
+void Task_7_bit_count() {
+  int number{0};
+
+  std::cout << "Enter number: ";
+  std::cin >> number;
+  std::cout << std::endl;
+
+  int checker{1};
+  int sum_of_bits{0};
+  for (int i = 0; i < sizeof(int) * 8; i++) {
+    if (number & checker)
+      sum_of_bits++;
+    checker <<= 1;
+  }
+
+  std::cout << "NUmber of set bits is: " << sum_of_bits << std::endl;
+}
+
+void Task_8_bit_set() {
+  int number{0};
+
+  std::cout << "Enter number: ";
+  std::cin >> number;
+  std::cout << std::endl;
+
+  int checker{0};
+  std::cout << "Enter bit number to check: ";
+  std::cin >> checker;
+  std::cout << std::endl;
+
+  constexpr int mask{1};
+  if (number & checker << mask)
```

Comment:
```suggestion
  if (number & (checker << 1))
```
в этом контексте 1 дасть больше понимания